### DISCUSSION TITLE
Uno project templates, difference between "Platform Library" and "Platform Cross-Runtime Library"

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno#18780] [General]
Hi

I'm currently creating a solution where it is needed to separate the code into several projects. When creating a new project there are both a "Uno Platform Library" and a Uno Platform Cross-Runtime Library" available. I have looked at the documentation and haven't been able to find anything regarding the difference for these ?
Is the Cross-runtime library a legacy from the .Net core days ? When trying to create it, it says to enable "Use previews of the .NET core SDK"

**Comment by jeromelaban** [comment]
Thanks for the question.

Cross-runtime libraries are an older version of Uno Platform Libraries when uno.sdk was not available. It's best to use Uno Platform Library projects now.

**Reply by natewaddoups** [reply]
It would be a good idea to include that information in the description of the project. The current experience for a new Uno user [criticism redacted].  As a new user myself, that was discouraging.

**Reply by jeromelaban** [reply]
Thanks for your insight. The easiest way to think about this is to not put intentions in the place of the developers building the platform. Still, we always welcome feedback on how to improve the user experience for newcomers.

We kept this template because it was still used by a portion of our user base, but as time passes, it does become less useful.

If you need more information about creating class libraries, [check out this doc](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html).

**Reply by MartinZikmund** [reply]
@natewaddoups @felixf4xu made PR to mark the old templates as `[Legacy]` - https://github.com/unoplatform/uno.studio/pull/704 , hopefully it will make things clearer 👍

**Reply by natewaddoups** [reply]
Thanks, I realize I'm really late here, but I do appreciate the quick turnaround on that.

**Comment by felixf4xu** [comment]
Is it possible to depredate it? I was trying to figure out the difference by comparing the description word by word yesterday.

![image](https://github.com/user-attachments/assets/fb36f962-0f83-4bd8-a61a-841e29c0fa57)

**Reply by MartinZikmund** [reply]
It should be marked as legacy in the latest releases
